### ISSUE TITLE
[JIT] Figure out how to easily investigate memory usage issues issues

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#27479]
e.g. https://github.com/pytorch/pytorch/issues/25267
 And other internal reports

**Comment by vadimkantorov** [comment]
also related to https://github.com/pytorch/pytorch/issues/1529

**Comment by vadimkantorov** [comment]
The master branch started to OOM because of variable sequence sizing when older commits seemed to work fine, so I have to pad more agressively (128 now, 16 was ok before). Also keeping more variable batch sizes causes severe slow-downs, I think something has changed in the memory allocator.

**Comment by colesbury** [comment]
@vadimkantorov are you referring to GPU memory? The referenced issue is CPU only.

**Comment by vadimkantorov** [comment]
@colesbury Yes, I'm referring to GPU memory. It seems that because of the sequence size issues, PyTorch now is susceptible to OOM

**Comment by jamesr66a** [comment]
cc @ilia-cher because of his memory profiling work

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
